### PR TITLE
Steps for Django email

### PR DESCRIPTION
**Pull request by RickyCook**

### CODE REVIEW

File: lettuce/django/steps/mail.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      ))
+def mail_sent_content(step, text, part):
+    """
+    Then I have sent an email with "pandas" in the body
+    """
+    for email in mail.outbox:
+        if text in getattr(email, part):
+            return True
+
+    assert False, "An email contained expected text in the {0}".format(part)
```
Comment (koterpillar) [inline lettuce/django/steps/mail.py]: `any(text in getattr(email, part) for email in mail.outbox)`

File: tests/integration/test_bamboo.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    assert_not_equals(status, 0)
+    assert "Length of outbox is 1" in out
+
+    FileSystem.popd()
+
+
+def test_mail_content():
+    'Mail content is checked through Lettuce steps'
+
+    FileSystem.pushd(current_directory, "django", "bamboo")
```
Comment (koterpillar) [inline tests/integration/test_bamboo.py]: Decorator, please?

```
@in_directory('django', 'bamboo'):
def test_xxxx():
    ....
```

File: tests/integration/django/bamboo/leaves/features/content.feature [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      "         Order ID: 10
+      "         Name: Fluffy Bear
+      "         Quantity: Quite a few
+      """
+
+    Then I have sent an email with the following in the body:
+      """
+      Name: Badger
+      Quantity: None
+      """
```
Comment (koterpillar) [inline tests/integration/django/bamboo/leaves/features/content.feature]: Blam.

**Comment by koterpillar** [comment]
Can you please ensure it meets at least our needs in, e.g. profile-server?

File: tests/integration/django/bamboo/leaves/features/content.feature (hunk 3 above)
Comment (RickyCook) [inline tests/integration/django/bamboo/leaves/features/content.feature]: ... what? :\

Comment (koterpillar) [inline tests/integration/django/bamboo/leaves/features/content.feature]: No newline at the end of file.

File: lettuce/django/steps/mail.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
+"""
+Step definitions for working with Django email.
+"""
+
+from django.core import mail
+
+from lettuce import step
+
+
+STEP_PREFIX = r'(?:Given|And|Then|When) '
```
Comment (koterpillar) [inline lettuce/django/steps/mail.py:10]: This is now defined both here and in `lettuce.django.steps.models`, move up?

File: tests/integration/django/bamboo/leaves/features/count.feature [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  Scenario: Passes if mail count is expected (single)
+    Given I send a test email
+    Then I have sent 1 email
+
+  Scenario: Passes if mail count is expected (multiple)
+    Given I send a test email
+    And I send a test email
+    Then I have sent 2 emails
+
+  # NEGATIVE TEST
```
Comment (koterpillar) [inline tests/integration/django/bamboo/leaves/features/count.feature:14]: Maybe use a tag for this?

File: tests/util.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+    cmd = 'python manage.py harvest -v 3 -T {0}{1}{2}{3}'.format(opts_string,
+                                                                 application,
+                                                                 feature,
+                                                                 scenario,
+                                                                 )
+    return commands.getstatusoutput(cmd)
+
+
+def in_directory(*directories):
```
Comment (koterpillar) [inline tests/util.py]: Should be a method on `FileSystem`.

Comment (RickyCook) [inline tests/util.py]: hmmm that's an idea

File: tests/integration/django/bamboo/leaves/features/content.feature [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+      "   - shipping@bamboodirect.com
+      " subject: New Order
+      " body: |
+      "         Order ID: 10
+      "         Name: Fluffy Bear
+      "         Quantity: Quite a few
+      """
+    And I send a test email with the following set:
+      """
+      " from_email: orders@bamboodirect.com
```
Comment (koterpillar) [inline tests/integration/django/bamboo/leaves/features/content.feature:38]: That's an actual site.

**Comment by RickyCook** [comment]
Raising PR for upstream

**Pull request closed without merging**